File: chat-service/src/main/java/com/kanban/chat/models/entities/Message.java
package com.kanban.chat.models.entities;

import com.kanban.chat.models.embedded.UserEmbedded;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Document(collection = "message")
public class Message {

    @Id
    private String id;
    private UserEmbedded sender;
    private String content;
    private Date instant;
    private MessageStatus status;
    @DBRef
    private Chat chat;
}

File: app-mobile/app/src/main/java/com/waly/chat/models/Message.kt
package com.waly.chat.models

import com.google.gson.annotations.SerializedName

data class Message(
    @SerializedName("id")
    val id: String,
    @SerializedName("sender")
    val sender: Sender,
    @SerializedName("content")
    val content: String,
    @SerializedName("instant")
    val instant: String
)